purchases/src/test/java/com/revenuecat/purchases/BackendTest.kt

```diff
@@ -606,7 +651,60 @@ class BackendTest {
             null,
             null,
             null,
-            true
+            true,
+            null,
+            null
+        )
+
+        assertThat(receivedPurchaserInfo).`as`("Received info is not null").isNotNull
+        assertThat(info).isEqualTo(receivedPurchaserInfo)
+    }
+
+    @Test
+    fun `postReceipt passes price and currency`() {
+        val info = postReceipt(
+            200,
+            false,
+            null,
+            null,
+            null,
+            true,
+            2.5,
+            "USD"
+        )
+
+        assertThat(receivedPurchaserInfo).`as`("Received info is not null").isNotNull
+        assertThat(info).isEqualTo(receivedPurchaserInfo)
+    }
+
+    @Test
+    fun `postReceipt passes null price and currency`() {
+        val info = postReceipt(
+            200,
+            false,
+            null,
+            null,
+            null,
+            true,
+            null,
+            null
+        )
+
+        assertThat(receivedPurchaserInfo).`as`("Received info is not null").isNotNull
+        assertThat(info).isEqualTo(receivedPurchaserInfo)
+    }
+
+    @Test
+    fun `given multiple post calls for same subscriber different price, both are triggered`() {
```

Comment:
is this test the same as `postReceipt passes null price and currency` and `postReceiptObserverMode`?

Reply:
Oopsies, thanks for catching this! It's true that `postReceipt passes null price and currency` and `postReceiptObserverMode` are the same. I removed `postReceipt passes null price and currency`